hoomd/md/methods/rattle.py

```diff
@@ -0,0 +1,463 @@
+# Copyright (c) 2009-2021 The Regents of the University of Michigan
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+"""MD integration methods."""
+
+from hoomd.md import _md
+import hoomd
+from hoomd.md.manifold import Manifold
+from hoomd.md.methods.methods import Method
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
+from hoomd.data.typeparam import TypeParameter
+from hoomd.data.typeconverter import OnlyTypes
+from hoomd.filter import ParticleFilter
+from hoomd.variant import Variant
+
+
+class MethodRATTLE(Method):
+    """Base class rattle integration method.
+
+    Provides common methods for all subclasses which are compatible with the
+    RATTLE algorithm. Particles can be constrained to a manifold surface using
+    the RATTLE scheme. For poor initial conditions that include overlapping
+    atoms, a limit can be specified to the movement a particle is allowed to
+    make in one time step. After a few thousand time steps with the limit set,
+    the system should be in a safe state to continue with unconstrained
+    integration.
+
+    Warning:
+        The particles should be initialised close to the implicit surface of
+        the manifold. Even though the particles are mapped to the set surface
+        automatically, the mapping can lead to small inter-particle distances
+        and, hence, large forces between particles!
+
+    For the equations of motion, see:
+
+    * S. Paquay and R. Kusters  2016
+      (`paper link <https://doi.org/10.1016/j.bpj.2016.02.017>`__)
```

Comment:
```suggestion
    See Also:

        `S. Paquay and R. Kusters  2016 <https://doi.org/10.1016/j.bpj.2016.02.017>`__
```
Change the link text to describe the content that the link points to.